Shared/LayerCacheManager.cpp

```diff
@@ -81,39 +82,26 @@ LayerCacheManager::LayerCacheManager(QObject* parent) :
   // obtain Add Local Data Controller
   m_localDataController = Toolkit::ToolManager::instance().tool<AddLocalDataController>();
 
-  if (m_localDataController)
+  // connect to new scene for cases where a new scene is set on the SceneView (via MobileScenePackage)
+  connect(Toolkit::ToolResourceProvider::instance(), &Toolkit::ToolResourceProvider::sceneChanged, this, [this]()
   {
-    // cache the created layers
-    connect(m_localDataController, &AddLocalDataController::layerCreated, this, [this](int layerIndex, Layer* layer)
-    {
-      m_initialLayerCache.insert(layerIndex, layer);
-      const int layerCount = m_inputLayerJsonArray.size();
-      emit jsonToLayerCompleted(layer);
+    m_scene = Toolkit::ToolResourceProvider::instance()->scene();
+    m_layers = QJsonArray();
+    m_inputLayerJsonArray = QJsonArray();
+    m_initialLayerCache.clear();
+    connectSignals();
+
+    // only add initial layers on initial load. Once the user selects a new scene, the layer list will be cleared
+    auto mobileSceneTool = Toolkit::ToolManager::instance().tool<OpenMobileScenePackageController>();
+    if (mobileSceneTool->userSelected())
```

Comment:
Is there any chance `mobileSceneTool` could be null?

Reply:
yes, perhaps. it's instantiated in a qml file so we can't exactly control order. i'll add a check